Assets/_Scripts/Tank/Tank MVC/TankView.cs

```diff
@@ -33,30 +27,53 @@ private void Update()
             {
                 tankController.FireBullet(transform.position, transform.eulerAngles);
             }
-
         }
 
         private void OnCollisionEnter(Collision collision)
         {
             if (collision.gameObject.GetComponent<BulletView>() != null)
             {
-                DestroyPlayer();
-                DestroyBullet();
+                StartCoroutine(HaltGameCoroutine());
             }
         }
 
-        private void DestroyBullet()
-        {
-            tankController.DestroyBullet();
-        }
-
         private void DestroyPlayer()
         {
-            Debug.Log("tank view destroyed");
+            //Debug.Log("tank view destroyed");
             tankController.DestroyController();
             Destroy(gameObject);
         }
 
+        private void VanishPlayer()
+        {
+            foreach (Renderer r in GetComponentsInChildren<Renderer>())
```

Comment:
get all components only once in the start/awake function, don't call again and again

Reply:
performance overhead